AutomatedTesting/Gem/PythonTests/largeworlds/gradient_signal/EditorScripts/ImageGradient_BilinearFiltering.py

```diff
@@ -0,0 +1,138 @@
+
+    # Set the Veg Spawner entity's Surface Tag Mask Filter component to include the "terrain" tag
+    hydra.get_set_test(spawner_entity, 3, "Configuration|Inclusion|Surface Tags", tag_list)
+
+    # Set the Image Gradient entity's Gradient Surface Tag Emitter component to modify the "terrain" tag
+    # NOTE: This requires a disable/re-enable of the component to force a refresh as assigning a tag via script does not
+    grad_surf_tag_emitter_component = image_gradient_entity.components[3]
+    grad_surf_tag_emitter_component.add_container_item("Configuration|Extended Tags", 0, tag_list[0])
+    grad_surf_tag_emitter_component.set_enabled(False)
+    grad_surf_tag_emitter_component.set_enabled(True)
+
+    # Assign the image asset to the image gradient and validate the expected asset was set
+    test_img_gradient_path = os.path.join("imagegradients", "largeworldsexamples", "out_height_gsi.tif.streamingimage")
+    asset = Asset.find_asset_by_path(test_img_gradient_path)
+    vegetation.ImageGradientRequestBus(bus.Event, "SetImageAssetPath", image_gradient_entity.id, test_img_gradient_path)
+    compare_asset_path = vegetation.ImageGradientRequestBus(bus.Event, "GetImageAssetPath", image_gradient_entity.id)
+    compare_asset = Asset.find_asset_by_path(compare_asset_path)
+    success = compare_asset.id == asset.id
+    Report.result(Tests.image_gradient_assigned, success)
+
+    # Update the Advanced settings for the Image Gradient component to use a lower resolution Mip
+    image_gradient_entity.components[0].set_component_property_value("Configuration|Advanced", True)
+    image_gradient_entity.components[0].set_component_property_value("Configuration|Advanced|Mip Index", 5)
+
+    # 5) Validate the expected number of vegetation instances when sampling via "Point"
+    # Re-assign the image gradient asset due to https://github.com/o3de/o3de/issues/9602
+    vegetation.ImageGradientRequestBus(bus.Event, "SetImageAssetPath", image_gradient_entity.id, test_img_gradient_path)
+    num_expected_instances_point = 1779
+    success = helper.wait_for_condition(lambda: dynveg.validate_instance_count_in_entity_shape(
+        spawner_entity.id, num_expected_instances_point), 5.0)
+    Report.result(Tests.point_instance_validation, success)
+
+    # 6) Validate the expected number of vegetation instances when sampling via "Bilinear"
+    image_gradient_entity.components[0].set_component_property_value("Configuration|Advanced|Sampling Type", 1)
+    # Re-assign the image gradient asset due to https://github.com/o3de/o3de/issues/9602
+    vegetation.ImageGradientRequestBus(bus.Event, "SetImageAssetPath", image_gradient_entity.id, test_img_gradient_path)
+    num_expected_instances_bilinear = 1839
+    success = helper.wait_for_condition(lambda: dynveg.validate_instance_count_in_entity_shape(
+        spawner_entity.id, num_expected_instances_bilinear), 5.0)
+    Report.result(Tests.bilinear_instance_validation, success)
```

Comment:
Is there a different way to configure the test such that there's a very explainable and predictable number of points being generated? If at some point in the future, this test failed because it generated 1841 points, but the points visually looked ok, I'd have a really hard time knowing what to do.

For example, you could create a test that plants 10 x 10 instances but only when the surface weight (or height or whatever) is > 0.1, and put the test box over an area that when point sampled would always return 0, but when bilinear sampled goes from 0 to 1, so 90% of the instances show up. Or something like that.